TaskModules/powershell/TaskModuleSqlUtility/SqlPackageOnTargetMachines.ps1

```diff
@@ -211,6 +211,45 @@ function LocateHighestVersionSqlPackageWithSql()
 
 function LocateHighestVersionSqlPackageWithDacMsi()
 {
+    $sqlDataTierFrameworkRegKeyWow = "HKLM:", "SOFTWARE", "Wow6432Node", "Microsoft", "Microsoft SQL Server", "Data-Tier Application Framework" -join [System.IO.Path]::DirectorySeparatorChar
+    $sqlDataTierFrameworkRegKey = "HKLM:", "SOFTWARE", "Microsoft", "Microsoft SQL Server", "Data-Tier Application Framework" -join [System.IO.Path]::DirectorySeparatorChar
+
+    if (-not (Test-Path $sqlDataTierFrameworkRegKeyWow))
+    {
+        $sqlDataTierFrameworkRegKeyWow = $sqlDataTierFrameworkRegKey
+    }
+
+    if ((Test-Path $sqlDataTierFrameworkRegKeyWow))
+    {
+        $keys = Get-Item $sqlDataTierFrameworkRegKeyWow | %{$_.GetSubKeyNames()} 
+        $versions = Get-SubKeysInFloatFormat $keys | Sort-Object -Descending
+
+        $installedMajorVersion = 0
+        foreach ($majorVersion in $versions)
+        {
+            $sqlInstallRootRegKey = "SOFTWARE", "Microsoft", "Microsoft SQL Server", "Data-Tier Application Framework", "$majorVersion" -join [System.IO.Path]::DirectorySeparatorChar
+            $sqlInstallRootPath64 = Get-RegistryValueIgnoreError LocalMachine "$sqlInstallRootRegKey" "InstallDir" Registry64
+            $sqlInstallRootPath32 = Get-RegistryValueIgnoreError LocalMachine "$sqlInstallRootRegKey" "InstallDir" Registry32
```

Comment:
You showed me offline that InstallDir is not available for versions below 130. But seems like you are trying to get InstallDir for all versions. I am assuming the logic here is - trying to get returns null and you will keep on continuing ultimately making this for loop a no-op. Am I right?